### ISSUE TITLE
Implement Initial LSTM Training Pipeline

### ISSUE THREAD

**Issue by nikhilc523** [nikhilc523/mobile-vision-training#16]
**Description:**

Implement LSTM training pipeline as specified in project proposal. Train initial model on proof-of-concept dataset (17 samples) to validate architecture and pipeline.

**Model Architecture (as per proposal):**
```python
model = keras.Sequential([
    Masking(mask_value=0.0, input_shape=(60, 6)),
    LSTM(64 units),
    Dropout(0.3),
    Dense(32, activation='relu'),
    Dense(1, activation='sigmoid')
])
```

**Tasks:**
- [x] Implement model architecture
- [x] Implement data loading and preprocessing
- [x] Implement train/val/test splitting
- [x] Implement class weighting
- [x] Implement data augmentation (time-warp, noise, dropout)
- [x] Implement early stopping
- [x] Implement metrics calculation (Precision, Recall, F1, ROC-AUC)
- [x] Create visualizations (training curves, ROC, confusion matrix)
- [x] Train on proof-of-concept dataset (17 samples)
- [x] Save model checkpoint
- [x] Document results

**Comment by nikhilc523** [comment]
**Results (Proof-of-Concept):**
- ✅ Model trained successfully
- ✅ Test F1: 0.857, Recall: 1.0, Precision: 0.75
- ✅ Model size: 79 KB (mobile-ready)
- ⚠️ Only 17 samples - not production-ready

**Next Steps:**
- [ ] Re-train on full dataset (~10,000+ samples) after Issue #7 complete
- [ ] Implement focal loss (Issue #10)
- [ ] Implement subject-wise splitting (Issue #11)
